ginga/table/TableView.py

```diff
@@ -0,0 +1,138 @@
+#
+# TableView.py -- Table viewer for Ginga
+#
+# This is open-source software licensed under a BSD license.
+# Please see the file LICENSE.txt for details.
+#
+from collections import OrderedDict
+
+from ginga.misc import Callback, Settings
+from ginga.misc import Bunch
+from ginga.util.six.moves import zip
+
+
+class TableViewBase(Callback.Callbacks):
+    """An abstract base class for displaying tables represented by
+    astropy table objects.
+
+    Parameters
+    ----------
+    logger : :py:class:`~logging.Logger` or `None`
+        Logger for tracing and debugging. If not given, one will be created.
+
+    settings : `~ginga.misc.Settings.SettingGroup` or `None`
+        Viewer preferences. If not given, one will be created.
+
+    """
+    def __init__(self, logger=None, settings=None):
+        Callback.Callbacks.__init__(self)
+
+        if logger is not None:
+            self.logger = logger
+        else:
+            self.logger = logging.Logger('TableViewBase')
+
+        # Create settings and set defaults
+        if settings is None:
+            settings = Settings.SettingGroup(logger=self.logger)
+        self.settings = settings
+
+        # for debugging
+        self.name = str(self)
+
+        self.settings.addDefaults(color_alternate_rows=True,
+                                  max_rows_for_col_resize=5000)
+
+        # For callbacks
+        for name in ('table-set', 'configure', ):
+            self.enable_callback(name)
+
+    def get_settings(self):
+        """Get the settings used by this instance.
+
+        Returns
+        -------
+        settings : `~ginga.misc.Settings.SettingGroup`
+            Settings.
+
+        """
+        return self.settings
+
+    def get_logger(self):
+        """Get the logger used by this instance.
+
+        Returns
+        -------
+        logger : :py:class:`~logging.Logger`
+            Logger.
+
+        """
+        return self.logger
+
+    def set_table(self, table):
+        self._table = table
+
+        self.make_callback('table-set', table)
+
+    def get_table(self):
+        return self._table
+
+    # for compatibility with ImageView
+    get_image = get_table
+
+
+class TableViewGw(TableViewBase):
+
+    def __init__(self, logger=None, settings=None):
+        super(TableViewGw, self).__init__(logger=logger, settings=settings)
+
+        self.add_callback('table-set', self.set_table_cb)
+
+        from ginga.gw import Widgets
+
+        # Create the viewer as a Treeview widget
+        color_alternate = self.settings.get('color_alternate_rows', True)
```

Comment:
It is not immediately clear how user can pass in such a setting to the table viewer. I mean, there is the keyword, but where is the `.cfg` file to use?